File: sdk/csharp-nancyfx/src/Org.OpenAPITools/Models/Item.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools.magicCashewbarcodable1_0_0.Models
{
    /// <summary>
    /// Item
    /// </summary>
    public sealed class Item:  IEquatable<Item>
    { 
        /// <summary>
        /// Upc
        /// </summary>
        public string Upc { get; private set; }

        /// <summary>
        /// Ean
        /// </summary>
        public string Ean { get; private set; }

        /// <summary>
        /// Isbn
        /// </summary>
        public string Isbn { get; private set; }

        /// <summary>
        /// Asin
        /// </summary>
        public string Asin { get; private set; }

        /// <summary>
        /// Title
        /// </summary>
        public string Title { get; private set; }

        /// <summary>
        /// Sku
        /// </summary>
        public string Sku { get; private set; }

        /// <summary>
        /// Mpn
        /// </summary>
        public string Mpn { get; private set; }

        /// <summary>
        /// PartNumber
        /// </summary>
        public string PartNumber { get; private set; }

        /// <summary>
        /// A list of matching upc codes.
        /// </summary>
        public List<string> Upcs { get; private set; }

        /// <summary>
        /// Description
        /// </summary>
        public string Description { get; private set; }

        /// <summary>
        /// Brand
        /// </summary>
        public string Brand { get; private set; }

        /// <summary>
        /// Manufacturer
        /// </summary>
        public string Manufacturer { get; private set; }

        /// <summary>
        /// Color
        /// </summary>
        public string Color { get; private set; }

        /// <summary>
        /// NewPrice
        /// </summary>
        public decimal? NewPrice { get; private set; }

        /// <summary>
        /// UsedPrice
        /// </summary>
        public decimal? UsedPrice { get; private set; }

        /// <summary>
        /// CurrencyCode
        /// </summary>
        public string CurrencyCode { get; private set; }

        /// <summary>
        /// Url
        /// </summary>
        public string Url { get; private set; }

        /// <summary>
        /// A list of product feature descriptions.
        /// </summary>
        public List<string> Features { get; private set; }

        /// <summary>
        /// Product dimensions and weight.  (&#39;meters&#39;, &#39;centimeters&#39;, &#39;millimeters&#39;, &#39;inches&#39;, &#39;feet&#39;, &#39;yards&#39;, &#39;ounces&#39;, &#39;pounds&#39;, &#39;grams&#39;, &#39;milligrams&#39;, &#39;kilograms&#39;)
        /// </summary>
        public List<Object> Dimensions { get; private set; }

        /// <summary>
        /// A list product images.
        /// </summary>
        public List<string> Images { get; private set; }

        /// <summary>
        /// Matched items for this product.
        /// </summary>
        public List<Object> MatchedItems { get; private set; }

        /// <summary>
        /// A list of country codes from which this item belongs.
        /// </summary>
        public List<string> IsoCountryCodes { get; private set; }

        /// <summary>
        /// The company name registered to the product&#39;s UPC or EAN.
        /// </summary>
        public string CompanyName { get; private set; }

        /// <summary>
        /// The company address registered to the product&#39;s UPC or EAN.
        /// </summary>
        public string CompanyAddress { get; private set; }

        /// <summary>
        /// A list of categories for this product.
        /// </summary>
        public List<string> Categories { get; private set; }

        /// <summary>
        /// A list of category hierarchies for this product. Categories are listed in order of generic category to most specific category.
        /// </summary>
        public List<string> CategoryHierarchies { get; private set; }


        /// <summary>
        /// Empty constructor required by some serializers.
        /// Use Item.Builder() for instance creation instead.
        /// </summary>
        [Obsolete]
        public Item()
        {
        }

        private Item(string Upc, string Ean, string Isbn, string Asin, string Title, string Sku, string Mpn, string PartNumber, List<string> Upcs, string Description, string Brand, string Manufacturer, string Color, decimal? NewPrice, decimal? UsedPrice, string CurrencyCode, string Url, List<string> Features, List<Object> Dimensions, List<string> Images, List<Object> MatchedItems, List<string> IsoCountryCodes, string CompanyName, string CompanyAddress, List<string> Categories, List<string> CategoryHierarchies)
        {
            
            this.Upc = Upc;
            
            this.Ean = Ean;
            
            this.Isbn = Isbn;
            
            this.Asin = Asin;
            
            this.Title = Title;
            
            this.Sku = Sku;
            
            this.Mpn = Mpn;
            
            this.PartNumber = PartNumber;
            
            this.Upcs = Upcs;
            
            this.Description = Description;
            
            this.Brand = Brand;
            
            this.Manufacturer = Manufacturer;
            
            this.Color = Color;
            
            this.NewPrice = NewPrice;
            
            this.UsedPrice = UsedPrice;
            
            this.CurrencyCode = CurrencyCode;
            
            this.Url = Url;
            
            this.Features = Features;
            
            this.Dimensions = Dimensions;
            
            this.Images = Images;
            
            this.MatchedItems = MatchedItems;
            
            this.IsoCountryCodes = IsoCountryCodes;
            
            this.CompanyName = CompanyName;
            
            this.CompanyAddress = CompanyAddress;
            
            this.Categories = Categories;
            
            this.CategoryHierarchies = CategoryHierarchies;
            
        }

        /// <summary>
        /// Returns builder of Item.
        /// </summary>
        /// <returns>ItemBuilder</returns>
        public static ItemBuilder Builder()
        {
            return new ItemBuilder();
        }

        /// <summary>
        /// Returns ItemBuilder with properties set.
        /// Use it to change properties.
        /// </summary>
        /// <returns>ItemBuilder</returns>
        public ItemBuilder With()
        {
            return Builder()
                .Upc(Upc)
                .Ean(Ean)
                .Isbn(Isbn)
                .Asin(Asin)
                .Title(Title)
                .Sku(Sku)
                .Mpn(Mpn)
                .PartNumber(PartNumber)
                .Upcs(Upcs)
                .Description(Description)
                .Brand(Brand)
                .Manufacturer(Manufacturer)
                .Color(Color)
                .NewPrice(NewPrice)
                .UsedPrice(UsedPrice)
                .CurrencyCode(CurrencyCode)
                .Url(Url)
                .Features(Features)
                .Dimensions(Dimensions)
                .Images(Images)
                .MatchedItems(MatchedItems)
                .IsoCountryCodes(IsoCountryCodes)
                .CompanyName(CompanyName)
                .CompanyAddress(CompanyAddress)
                .Categories(Categories)
                .CategoryHierarchies(CategoryHierarchies);
        }

        public override string ToString()
        {
            return this.PropertiesToString();
        }

        public override bool Equals(object obj)
        {
            return this.EqualsByProperties(obj);
        }

        public bool Equals(Item other)
        {
            return Equals((object) other);
        }

        public override int GetHashCode()
        {
            return this.PropertiesHash();
        }

        /// <summary>
        /// Implementation of == operator for (Item.
        /// </summary>
        /// <param name="left">Compared (Item</param>
        /// <param name="right">Compared (Item</param>
        /// <returns>true if compared items are equals, false otherwise</returns>
        public static bool operator == (Item left, Item right)
        {
            return Equals(left, right);
        }

        /// <summary>
        /// Implementation of != operator for (Item.
        /// </summary>
        /// <param name="left">Compared (Item</param>
        /// <param name="right">Compared (Item</param>
        /// <returns>true if compared items are not equals, false otherwise</returns>
        public static bool operator != (Item left, Item right)
        {
            return !Equals(left, right);
        }

        /// <summary>
        /// Builder of Item.
        /// </summary>
        public sealed class ItemBuilder
        {
            private string _Upc;
            private string _Ean;
            private string _Isbn;
            private string _Asin;
            private string _Title;
            private string _Sku;
            private string _Mpn;
            private string _PartNumber;
            private List<string> _Upcs;
            private string _Description;
            private string _Brand;
            private string _Manufacturer;
            private string _Color;
            private decimal? _NewPrice;
            private decimal? _UsedPrice;
            private string _CurrencyCode;
            private string _Url;
            private List<string> _Features;
            private List<Object> _Dimensions;
            private List<string> _Images;
            private List<Object> _MatchedItems;
            private List<string> _IsoCountryCodes;
            private string _CompanyName;
            private string _CompanyAddress;
            private List<string> _Categories;
            private List<string> _CategoryHierarchies;

            internal ItemBuilder()
            {
                SetupDefaults();
            }

            private void SetupDefaults()
            {
            }

            /// <summary>
            /// Sets value for Item.Upc property.
            /// </summary>
            /// <param name="value">Upc</param>
            public ItemBuilder Upc(string value)
            {
                _Upc = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Ean property.
            /// </summary>
            /// <param name="value">Ean</param>
            public ItemBuilder Ean(string value)
            {
                _Ean = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Isbn property.
            /// </summary>
            /// <param name="value">Isbn</param>
            public ItemBuilder Isbn(string value)
            {
                _Isbn = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Asin property.
            /// </summary>
            /// <param name="value">Asin</param>
            public ItemBuilder Asin(string value)
            {
                _Asin = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Title property.
            /// </summary>
            /// <param name="value">Title</param>
            public ItemBuilder Title(string value)
            {
                _Title = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Sku property.
            /// </summary>
            /// <param name="value">Sku</param>
            public ItemBuilder Sku(string value)
            {
                _Sku = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Mpn property.
            /// </summary>
            /// <param name="value">Mpn</param>
            public ItemBuilder Mpn(string value)
            {
                _Mpn = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.PartNumber property.
            /// </summary>
            /// <param name="value">PartNumber</param>
            public ItemBuilder PartNumber(string value)
            {
                _PartNumber = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Upcs property.
            /// </summary>
            /// <param name="value">A list of matching upc codes.</param>
            public ItemBuilder Upcs(List<string> value)
            {
                _Upcs = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Description property.
            /// </summary>
            /// <param name="value">Description</param>
            public ItemBuilder Description(string value)
            {
                _Description = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Brand property.
            /// </summary>
            /// <param name="value">Brand</param>
            public ItemBuilder Brand(string value)
            {
                _Brand = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Manufacturer property.
            /// </summary>
            /// <param name="value">Manufacturer</param>
            public ItemBuilder Manufacturer(string value)
            {
                _Manufacturer = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Color property.
            /// </summary>
            /// <param name="value">Color</param>
            public ItemBuilder Color(string value)
            {
                _Color = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.NewPrice property.
            /// </summary>
            /// <param name="value">NewPrice</param>
            public ItemBuilder NewPrice(decimal? value)
            {
                _NewPrice = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.UsedPrice property.
            /// </summary>
            /// <param name="value">UsedPrice</param>
            public ItemBuilder UsedPrice(decimal? value)
            {
                _UsedPrice = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.CurrencyCode property.
            /// </summary>
            /// <param name="value">CurrencyCode</param>
            public ItemBuilder CurrencyCode(string value)
            {
                _CurrencyCode = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Url property.
            /// </summary>
            /// <param name="value">Url</param>
            public ItemBuilder Url(string value)
            {
                _Url = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Features property.
            /// </summary>
            /// <param name="value">A list of product feature descriptions.</param>
            public ItemBuilder Features(List<string> value)
            {
                _Features = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Dimensions property.
            /// </summary>
            /// <param name="value">Product dimensions and weight.  (&#39;meters&#39;, &#39;centimeters&#39;, &#39;millimeters&#39;, &#39;inches&#39;, &#39;feet&#39;, &#39;yards&#39;, &#39;ounces&#39;, &#39;pounds&#39;, &#39;grams&#39;, &#39;milligrams&#39;, &#39;kilograms&#39;)</param>
            public ItemBuilder Dimensions(List<Object> value)
            {
                _Dimensions = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Images property.
            /// </summary>
            /// <param name="value">A list product images.</param>
            public ItemBuilder Images(List<string> value)
            {
                _Images = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.MatchedItems property.
            /// </summary>
            /// <param name="value">Matched items for this product.</param>
            public ItemBuilder MatchedItems(List<Object> value)
            {
                _MatchedItems = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.IsoCountryCodes property.
            /// </summary>
            /// <param name="value">A list of country codes from which this item belongs.</param>
            public ItemBuilder IsoCountryCodes(List<string> value)
            {
                _IsoCountryCodes = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.CompanyName property.
            /// </summary>
            /// <param name="value">The company name registered to the product&#39;s UPC or EAN.</param>
            public ItemBuilder CompanyName(string value)
            {
                _CompanyName = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.CompanyAddress property.
            /// </summary>
            /// <param name="value">The company address registered to the product&#39;s UPC or EAN.</param>
            public ItemBuilder CompanyAddress(string value)
            {
                _CompanyAddress = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.Categories property.
            /// </summary>
            /// <param name="value">A list of categories for this product.</param>
            public ItemBuilder Categories(List<string> value)
            {
                _Categories = value;
                return this;
            }

            /// <summary>
            /// Sets value for Item.CategoryHierarchies property.
            /// </summary>
            /// <param name="value">A list of category hierarchies for this product. Categories are listed in order of generic category to most specific category.</param>
            public ItemBuilder CategoryHierarchies(List<string> value)
            {
                _CategoryHierarchies = value;
                return this;
            }


            /// <summary>
            /// Builds instance of Item.
            /// </summary>
            /// <returns>Item</returns>
            public Item Build()
            {
                Validate();
                return new Item(
                    Upc: _Upc,
                    Ean: _Ean,
                    Isbn: _Isbn,
                    Asin: _Asin,
                    Title: _Title,
                    Sku: _Sku,
                    Mpn: _Mpn,
                    PartNumber: _PartNumber,
                    Upcs: _Upcs,
                    Description: _Description,
                    Brand: _Brand,
                    Manufacturer: _Manufacturer,
                    Color: _Color,
                    NewPrice: _NewPrice,
                    UsedPrice: _UsedPrice,
                    CurrencyCode: _CurrencyCode,
                    Url: _Url,
                    Features: _Features,
                    Dimensions: _Dimensions,
                    Images: _Images,
                    MatchedItems: _MatchedItems,
                    IsoCountryCodes: _IsoCountryCodes,
                    CompanyName: _CompanyName,
                    CompanyAddress: _CompanyAddress,
                    Categories: _Categories,
                    CategoryHierarchies: _CategoryHierarchies
                );
            }

            private void Validate()
            { 
            }
        }

        
    }
}

File: sdk/java-play-framework/app/apimodels/Item.java
package apimodels;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * Item
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2020-03-27T18:10:03.884-04:00[America/New_York]")

@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class Item   {
  @JsonProperty("upc")
  private String upc;

  @JsonProperty("ean")
  private String ean;

  @JsonProperty("isbn")
  private String isbn;

  @JsonProperty("asin")
  private String asin;

  @JsonProperty("title")
  private String title;

  @JsonProperty("sku")
  private String sku;

  @JsonProperty("mpn")
  private String mpn;

  @JsonProperty("part_number")
  private String partNumber;

  @JsonProperty("upcs")
  private List<String> upcs = null;

  @JsonProperty("description")
  private String description;

  @JsonProperty("brand")
  private String brand;

  @JsonProperty("manufacturer")
  private String manufacturer;

  @JsonProperty("color")
  private String color;

  @JsonProperty("new_price")
  private BigDecimal newPrice;

  @JsonProperty("used_price")
  private BigDecimal usedPrice;

  @JsonProperty("currency_code")
  private String currencyCode;

  @JsonProperty("url")
  private String url;

  @JsonProperty("features")
  private List<String> features = null;

  @JsonProperty("dimensions")
  private List<Object> dimensions = null;

  @JsonProperty("images")
  private List<String> images = null;

  @JsonProperty("matched_items")
  private List<Object> matchedItems = null;

  @JsonProperty("iso_country_codes")
  private List<String> isoCountryCodes = null;

  @JsonProperty("company_name")
  private String companyName;

  @JsonProperty("company_address")
  private String companyAddress;

  @JsonProperty("categories")
  private List<String> categories = null;

  @JsonProperty("category_hierarchies")
  private List<String> categoryHierarchies = null;

  public Item upc(String upc) {
    this.upc = upc;
    return this;
  }

   /**
   * Get upc
   * @return upc
  **/
    public String getUpc() {
    return upc;
  }

  public void setUpc(String upc) {
    this.upc = upc;
  }

  public Item ean(String ean) {
    this.ean = ean;
    return this;
  }

   /**
   * Get ean
   * @return ean
  **/
    public String getEan() {
    return ean;
  }

  public void setEan(String ean) {
    this.ean = ean;
  }

  public Item isbn(String isbn) {
    this.isbn = isbn;
    return this;
  }

   /**
   * Get isbn
   * @return isbn
  **/
    public String getIsbn() {
    return isbn;
  }

  public void setIsbn(String isbn) {
    this.isbn = isbn;
  }

  public Item asin(String asin) {
    this.asin = asin;
    return this;
  }

   /**
   * Get asin
   * @return asin
  **/
    public String getAsin() {
    return asin;
  }

  public void setAsin(String asin) {
    this.asin = asin;
  }

  public Item title(String title) {
    this.title = title;
    return this;
  }

   /**
   * Get title
   * @return title
  **/
    public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public Item sku(String sku) {
    this.sku = sku;
    return this;
  }

   /**
   * Get sku
   * @return sku
  **/
    public String getSku() {
    return sku;
  }

  public void setSku(String sku) {
    this.sku = sku;
  }

  public Item mpn(String mpn) {
    this.mpn = mpn;
    return this;
  }

   /**
   * Get mpn
   * @return mpn
  **/
    public String getMpn() {
    return mpn;
  }

  public void setMpn(String mpn) {
    this.mpn = mpn;
  }

  public Item partNumber(String partNumber) {
    this.partNumber = partNumber;
    return this;
  }

   /**
   * Get partNumber
   * @return partNumber
  **/
    public String getPartNumber() {
    return partNumber;
  }

  public void setPartNumber(String partNumber) {
    this.partNumber = partNumber;
  }

  public Item upcs(List<String> upcs) {
    this.upcs = upcs;
    return this;
  }

  public Item addUpcsItem(String upcsItem) {
    if (upcs == null) {
      upcs = new ArrayList<>();
    }
    upcs.add(upcsItem);
    return this;
  }

   /**
   * A list of matching upc codes.
   * @return upcs
  **/
    public List<String> getUpcs() {
    return upcs;
  }

  public void setUpcs(List<String> upcs) {
    this.upcs = upcs;
  }

  public Item description(String description) {
    this.description = description;
    return this;
  }

   /**
   * Get description
   * @return description
  **/
    public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public Item brand(String brand) {
    this.brand = brand;
    return this;
  }

   /**
   * Get brand
   * @return brand
  **/
    public String getBrand() {
    return brand;
  }

  public void setBrand(String brand) {
    this.brand = brand;
  }

  public Item manufacturer(String manufacturer) {
    this.manufacturer = manufacturer;
    return this;
  }

   /**
   * Get manufacturer
   * @return manufacturer
  **/
    public String getManufacturer() {
    return manufacturer;
  }

  public void setManufacturer(String manufacturer) {
    this.manufacturer = manufacturer;
  }

  public Item color(String color) {
    this.color = color;
    return this;
  }

   /**
   * Get color
   * @return color
  **/
    public String getColor() {
    return color;
  }

  public void setColor(String color) {
    this.color = color;
  }

  public Item newPrice(BigDecimal newPrice) {
    this.newPrice = newPrice;
    return this;
  }

   /**
   * Get newPrice
   * @return newPrice
  **/
  @Valid
  public BigDecimal getNewPrice() {
    return newPrice;
  }

  public void setNewPrice(BigDecimal newPrice) {
    this.newPrice = newPrice;
  }

  public Item usedPrice(BigDecimal usedPrice) {
    this.usedPrice = usedPrice;
    return this;
  }

   /**
   * Get usedPrice
   * @return usedPrice
  **/
  @Valid
  public BigDecimal getUsedPrice() {
    return usedPrice;
  }

  public void setUsedPrice(BigDecimal usedPrice) {
    this.usedPrice = usedPrice;
  }

  public Item currencyCode(String currencyCode) {
    this.currencyCode = currencyCode;
    return this;
  }

   /**
   * Get currencyCode
   * @return currencyCode
  **/
    public String getCurrencyCode() {
    return currencyCode;
  }

  public void setCurrencyCode(String currencyCode) {
    this.currencyCode = currencyCode;
  }

  public Item url(String url) {
    this.url = url;
    return this;
  }

   /**
   * Get url
   * @return url
  **/
    public String getUrl() {
    return url;
  }

  public void setUrl(String url) {
    this.url = url;
  }

  public Item features(List<String> features) {
    this.features = features;
    return this;
  }

  public Item addFeaturesItem(String featuresItem) {
    if (features == null) {
      features = new ArrayList<>();
    }
    features.add(featuresItem);
    return this;
  }

   /**
   * A list of product feature descriptions.
   * @return features
  **/
    public List<String> getFeatures() {
    return features;
  }

  public void setFeatures(List<String> features) {
    this.features = features;
  }

  public Item dimensions(List<Object> dimensions) {
    this.dimensions = dimensions;
    return this;
  }

  public Item addDimensionsItem(Object dimensionsItem) {
    if (dimensions == null) {
      dimensions = new ArrayList<>();
    }
    dimensions.add(dimensionsItem);
    return this;
  }

   /**
   * Product dimensions and weight.  ('meters', 'centimeters', 'millimeters', 'inches', 'feet', 'yards', 'ounces', 'pounds', 'grams', 'milligrams', 'kilograms')
   * @return dimensions
  **/
    public List<Object> getDimensions() {
    return dimensions;
  }

  public void setDimensions(List<Object> dimensions) {
    this.dimensions = dimensions;
  }

  public Item images(List<String> images) {
    this.images = images;
    return this;
  }

  public Item addImagesItem(String imagesItem) {
    if (images == null) {
      images = new ArrayList<>();
    }
    images.add(imagesItem);
    return this;
  }

   /**
   * A list product images.
   * @return images
  **/
    public List<String> getImages() {
    return images;
  }

  public void setImages(List<String> images) {
    this.images = images;
  }

  public Item matchedItems(List<Object> matchedItems) {
    this.matchedItems = matchedItems;
    return this;
  }

  public Item addMatchedItemsItem(Object matchedItemsItem) {
    if (matchedItems == null) {
      matchedItems = new ArrayList<>();
    }
    matchedItems.add(matchedItemsItem);
    return this;
  }

   /**
   * Matched items for this product.
   * @return matchedItems
  **/
    public List<Object> getMatchedItems() {
    return matchedItems;
  }

  public void setMatchedItems(List<Object> matchedItems) {
    this.matchedItems = matchedItems;
  }

  public Item isoCountryCodes(List<String> isoCountryCodes) {
    this.isoCountryCodes = isoCountryCodes;
    return this;
  }

  public Item addIsoCountryCodesItem(String isoCountryCodesItem) {
    if (isoCountryCodes == null) {
      isoCountryCodes = new ArrayList<>();
    }
    isoCountryCodes.add(isoCountryCodesItem);
    return this;
  }

   /**
   * A list of country codes from which this item belongs.
   * @return isoCountryCodes
  **/
    public List<String> getIsoCountryCodes() {
    return isoCountryCodes;
  }

  public void setIsoCountryCodes(List<String> isoCountryCodes) {
    this.isoCountryCodes = isoCountryCodes;
  }

  public Item companyName(String companyName) {
    this.companyName = companyName;
    return this;
  }

   /**
   * The company name registered to the product's UPC or EAN.
   * @return companyName
  **/
    public String getCompanyName() {
    return companyName;
  }

  public void setCompanyName(String companyName) {
    this.companyName = companyName;
  }

  public Item companyAddress(String companyAddress) {
    this.companyAddress = companyAddress;
    return this;
  }

   /**
   * The company address registered to the product's UPC or EAN.
   * @return companyAddress
  **/
    public String getCompanyAddress() {
    return companyAddress;
  }

  public void setCompanyAddress(String companyAddress) {
    this.companyAddress = companyAddress;
  }

  public Item categories(List<String> categories) {
    this.categories = categories;
    return this;
  }

  public Item addCategoriesItem(String categoriesItem) {
    if (categories == null) {
      categories = new ArrayList<>();
    }
    categories.add(categoriesItem);
    return this;
  }

   /**
   * A list of categories for this product.
   * @return categories
  **/
    public List<String> getCategories() {
    return categories;
  }

  public void setCategories(List<String> categories) {
    this.categories = categories;
  }

  public Item categoryHierarchies(List<String> categoryHierarchies) {
    this.categoryHierarchies = categoryHierarchies;
    return this;
  }

  public Item addCategoryHierarchiesItem(String categoryHierarchiesItem) {
    if (categoryHierarchies == null) {
      categoryHierarchies = new ArrayList<>();
    }
    categoryHierarchies.add(categoryHierarchiesItem);
    return this;
  }

   /**
   * A list of category hierarchies for this product. Categories are listed in order of generic category to most specific category.
   * @return categoryHierarchies
  **/
    public List<String> getCategoryHierarchies() {
    return categoryHierarchies;
  }

  public void setCategoryHierarchies(List<String> categoryHierarchies) {
    this.categoryHierarchies = categoryHierarchies;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Item item = (Item) o;
    return Objects.equals(upc, item.upc) &&
        Objects.equals(ean, item.ean) &&
        Objects.equals(isbn, item.isbn) &&
        Objects.equals(asin, item.asin) &&
        Objects.equals(title, item.title) &&
        Objects.equals(sku, item.sku) &&
        Objects.equals(mpn, item.mpn) &&
        Objects.equals(partNumber, item.partNumber) &&
        Objects.equals(upcs, item.upcs) &&
        Objects.equals(description, item.description) &&
        Objects.equals(brand, item.brand) &&
        Objects.equals(manufacturer, item.manufacturer) &&
        Objects.equals(color, item.color) &&
        Objects.equals(newPrice, item.newPrice) &&
        Objects.equals(usedPrice, item.usedPrice) &&
        Objects.equals(currencyCode, item.currencyCode) &&
        Objects.equals(url, item.url) &&
        Objects.equals(features, item.features) &&
        Objects.equals(dimensions, item.dimensions) &&
        Objects.equals(images, item.images) &&
        Objects.equals(matchedItems, item.matchedItems) &&
        Objects.equals(isoCountryCodes, item.isoCountryCodes) &&
        Objects.equals(companyName, item.companyName) &&
        Objects.equals(companyAddress, item.companyAddress) &&
        Objects.equals(categories, item.categories) &&
        Objects.equals(categoryHierarchies, item.categoryHierarchies);
  }

  @Override
  public int hashCode() {
    return Objects.hash(upc, ean, isbn, asin, title, sku, mpn, partNumber, upcs, description, brand, manufacturer, color, newPrice, usedPrice, currencyCode, url, features, dimensions, images, matchedItems, isoCountryCodes, companyName, companyAddress, categories, categoryHierarchies);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Item {\n");
    
    sb.append("    upc: ").append(toIndentedString(upc)).append("\n");
    sb.append("    ean: ").append(toIndentedString(ean)).append("\n");
    sb.append("    isbn: ").append(toIndentedString(isbn)).append("\n");
    sb.append("    asin: ").append(toIndentedString(asin)).append("\n");
    sb.append("    title: ").append(toIndentedString(title)).append("\n");
    sb.append("    sku: ").append(toIndentedString(sku)).append("\n");
    sb.append("    mpn: ").append(toIndentedString(mpn)).append("\n");
    sb.append("    partNumber: ").append(toIndentedString(partNumber)).append("\n");
    sb.append("    upcs: ").append(toIndentedString(upcs)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    brand: ").append(toIndentedString(brand)).append("\n");
    sb.append("    manufacturer: ").append(toIndentedString(manufacturer)).append("\n");
    sb.append("    color: ").append(toIndentedString(color)).append("\n");
    sb.append("    newPrice: ").append(toIndentedString(newPrice)).append("\n");
    sb.append("    usedPrice: ").append(toIndentedString(usedPrice)).append("\n");
    sb.append("    currencyCode: ").append(toIndentedString(currencyCode)).append("\n");
    sb.append("    url: ").append(toIndentedString(url)).append("\n");
    sb.append("    features: ").append(toIndentedString(features)).append("\n");
    sb.append("    dimensions: ").append(toIndentedString(dimensions)).append("\n");
    sb.append("    images: ").append(toIndentedString(images)).append("\n");
    sb.append("    matchedItems: ").append(toIndentedString(matchedItems)).append("\n");
    sb.append("    isoCountryCodes: ").append(toIndentedString(isoCountryCodes)).append("\n");
    sb.append("    companyName: ").append(toIndentedString(companyName)).append("\n");
    sb.append("    companyAddress: ").append(toIndentedString(companyAddress)).append("\n");
    sb.append("    categories: ").append(toIndentedString(categories)).append("\n");
    sb.append("    categoryHierarchies: ").append(toIndentedString(categoryHierarchies)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

